spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1180,11 +1180,18 @@ def _run_algorithms(
             self._machine_allocation_controller = executor.get_item(
                 "MachineAllocationController")
             report_folder = executor.get_item("ReportFolder")
-            TagsFromMachineReport()(report_folder, self._txrx)
+            try:
+                if report_folder:
+                    TagsFromMachineReport()(report_folder, self._txrx)
+            except Exception as e2:
+                logger.warning(
+                    "problem with TagsFromMachineReport {}".format(e2),
+                    exc_info=True)
             try:
                 self._shutdown()
-            except Exception:
-                logger.warning("problem when shutting down", exc_info=True)
+            except Exception as e3:
+                logger.warning("problem when shutting down {}".format(e3),
+                               exc_info=True)
```

Comment:
Technically, I believe the exc_info=True means that the exception details are displayed anyway, but this isn't critical.

Reply:
This way also adds it to the log message repeat at the very end.

Reply:
OK, fair enough